rest/replication_api_test.go

```diff
@@ -1764,3 +1764,123 @@ func TestDBReplicationStatsTeardown(t *testing.T) {
 	require.NoError(t, err)
 
 }
+
+// Repros CBG-2450 - Make sure doc that have an ID prefixed with a null byte don't get sent over BLIP
+func TestReplicateDocIDNullCharPrefix(t *testing.T) {
```

Comment:
I'm wondering whether we can change (or add) a changes feed unit-test (in the db package)

It's more targeted, and avoids the overhead of 2 buckets/RestTester/ISGR and could potentially give us easier coverage of the more risky areas of this change like around limits/sequence numbering if we decide to go for the extra coverage.